config/backend-config/multitenant_workspaces.go

```diff
@@ -136,41 +139,46 @@ func (workspaceConfig *MultiTenantWorkspacesConfig) getFromAPI(ctx context.Conte
 		}
 		sourcesJSON.Sources = append(sourcesJSON.Sources, workspaceConfig.Sources...)
 	}
-	workspaceConfig.workspaceWriteKeysMapLock.Lock()
-	workspaceConfig.writeKeyToWorkspaceIDMap = writeKeyToWorkspaceIDMap
-	workspaceConfig.sourceToWorkspaceIDMap = sourceToWorkspaceIDMap
-	workspaceConfig.workspaceIDToLibrariesMap = workspaceIDToLibrariesMap
-	workspaceConfig.workspaceWriteKeysMapLock.Unlock()
+	wc.workspaceWriteKeysMapLock.Lock()
+	wc.writeKeyToWorkspaceIDMap = writeKeyToWorkspaceIDMap
+	wc.sourceToWorkspaceIDMap = sourceToWorkspaceIDMap
+	wc.workspaceIDToLibrariesMap = workspaceIDToLibrariesMap
+	wc.workspaceWriteKeysMapLock.Unlock()
 
 	return sourcesJSON, nil
 }
 
-func (workspaceConfig *MultiTenantWorkspacesConfig) makeHTTPRequest(
+func (wc *multiTenantWorkspacesConfig) makeHTTPRequest(
 	ctx context.Context, url string,
 ) ([]byte, int, error) {
-	req, err := Http.NewRequestWithContext(ctx, "GET", url, http.NoBody)
+	req, err := http.NewRequestWithContext(ctx, "GET", url, http.NoBody)
 	if err != nil {
 		return []byte{}, 400, err
 	}
 	// TODO: hacky way to get the backend config for multi tenant through older hosted backend config
 	if config.GetBool("BackendConfig.useHostedBackendConfig", false) {
 		req.SetBasicAuth(config.GetEnv("HOSTED_SERVICE_SECRET", ""), "")
 	} else {
-		req.SetBasicAuth(workspaceConfig.Token, "")
+		req.SetBasicAuth(wc.Token, "")
 	}
 
 	req.Header.Set("Content-Type", "application/json")
 
 	client := &http.Client{Timeout: config.GetDuration("HttpClient.timeout", 30, time.Second)}
 	resp, err := client.Do(req)
 	if err != nil {
-		return []byte{}, 400, err
+		return nil, http.StatusBadRequest, err
+	}
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, http.StatusBadRequest, err
 	}
 
-	var respBody []byte
-	if resp != nil && resp.Body != nil {
-		respBody, _ = IoUtil.ReadAll(resp.Body)
-		defer func() { _ = resp.Body.Close() }()
+	defer func() { _ = resp.Body.Close() }()
+
+	if resp.StatusCode >= 300 {
```

Comment:
since we appear to be performing error logic based on the status code here, does it make sense to remove the status code from this method's return values too?

Reply:
yes!